packages/headless/src/features/commerce/recommendations/recommendations.ts

```diff
@@ -1,9 +1,13 @@
 import {Schema} from '@coveo/bueno';
 import {RecommendationsOptions} from '../../../controllers/commerce/recommendations/headless-recommendations';
-import {requiredNonEmptyString} from '../../../utils/validate-payload';
+import {
+  nonRequiredEmptyAllowedString,
+  requiredNonEmptyString,
+} from '../../../utils/validate-payload';
 
 export const recommendationsSlotDefinition = {
   slotId: requiredNonEmptyString,
+  productId: nonRequiredEmptyAllowedString,
```

Comment:
Is it really okay to pass in `productId=""` ?

Should it be this instead ?

```typescript
new StringValue({required: false, emptyAllowed: false})
```

If it's okay and a legit use case, disregard this comment

Reply:
It just does nothing, so empty string is no harm done

Reply:
We should prefer being stricter in that case, and apply Oli's suggestion